ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm_step1/device/moreh_clip_grad_norm_step1_program_factory.cpp

```diff
@@ -0,0 +1,222 @@
+            {tt::CB::c_intermed1, im1_t},  // |x|^p
+            {tt::CB::c_intermed2, im2_t},  // Add[|x|^p * exp(log(|x|) * decimal)]
+            {tt::CB::c_intermed3, im3_t},  // log(|x|)
+            {tt::CB::c_intermed4, im4_t},  // exp(log(|x|) * decimal)
+            {tt::CB::c_intermed5, im5_t},  // |x|^p * exp(log(|x|) * decimal)
+        });
+
+    ////////////////////////////////////////////////////////////////////////////
+    //                      DataMovementKernel SetUp
+    ////////////////////////////////////////////////////////////////////////////
+    const auto reader_kernel_file =
+        "ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm_step1/device/kernels/"
+        "reader_moreh_clip_grad_norm_step1.cpp";
+    const auto writer_kernel_file =
+        "ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm_step1/device/kernels/"
+        "writer_moreh_clip_grad_norm_step1.cpp";
+
+    const auto reader_kernel_id = ttnn::operations::CreateReadKernel(program, reader_kernel_file, core_group_1);
+    const auto writer_kernel_id = ttnn::operations::CreateWriteKernel(program, writer_kernel_file, core_group_1);
+
+    ////////////////////////////////////////////////////////////////////////////
+    //                      ComputeKernel SetUp
+    ////////////////////////////////////////////////////////////////////////////
+    std::map<std::string, std::string> compute_defines{};
+    compute_defines["REDUCE_OP"] = "PoolType::SUM";
+    compute_defines["REDUCE_DIM"] = "ReduceDim::REDUCE_SCALAR";
+
+    const auto compute_kernel_file =
+        "ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm_step1/device/kernels/"
+        "moreh_clip_grad_norm_step1_kernel.cpp";
+
+    const auto compute_kernel_id = ttnn::operations::CreateComputeKernel(
+        program, compute_kernel_file, {core_group_1, num_inputs_per_core_group_1}, compute_defines);
+
+    ////////////////////////////////////////////////////////////////////////////
+    //                      RuntimeArgs SetUp
+    ////////////////////////////////////////////////////////////////////////////
+
+    const auto output_addr = tmp_pow_sum.buffer()->address();
+    bool tmp_pow_sum_is_dram = tmp_pow_sum.buffer()->buffer_type() == tt::tt_metal::BufferType::DRAM ? 1 : 0;
+
+    uint32_t tile_offset = tile_offset_of_tmp_pow_sum;
+    for (uint32_t i = 0; i < num_cores_to_be_used; ++i) {
+        CoreCoord core = {i / num_cores_y, i % num_cores_y};
+
+        const auto& input = inputs.at(i);
+        const auto input_addr = input.buffer()->address();
+        const auto num_tiles = input.volume() / tt::constants::TILE_HW;
+        const auto [origin_h, origin_w] = origin_hw_vec.at(i);
+
+        // reader
+        const std::array reader_runtime_args{
+            input_addr,
+            static_cast<uint32_t>(input.buffer()->buffer_type() == tt::tt_metal::BufferType::DRAM ? 1 : 0),
```

Comment:
Similar to above

Reply:
Resolved in [367b454](https://github.com/tenstorrent/tt-metal/pull/14667/commits/367b454f49e71941dc71e9cf323bf305bcb799ba)